### PR TITLE
ITM-476

### PR DESCRIPTION
**Pull request by Ayal-Ciobotaru**
Reworked spo2 from a float to normalized VOCAB based on the following

- NORMAL: A blood oxygen level of 92 or higher. A normal level of oxygen is usually 95% or higher. Some people with chronic lung disease or sleep apnea can have normal levels around 90%.

- LOW: A blood oxygen level 85 or higher but less than 92.

- NONE: With severe Hypoxia (below 85), the pulse oximeter can become unreliable and blood oxygen level may be undetectable. The patient could be pulseless or not oxygenating at all.

### CODE REVIEW

File: swagger/swagger.yaml [hunk 1]
```diff
@@ -1906,6 +1902,13 @@ components:
         - SEARCH
         - SITREP
         - TAG_CHARACTER
+    BloodOxygenEnum:
+      type: string
+      description: The percentage of oxygen in someone's blood, as measured by a pulse oximeter
```
Comment (nextcen-dgemoets) [inline swagger/swagger.yaml:1907]: Although blood ox is a percentage, this enum really isn't.  How about "A description of the percentage of oxygen in someone's blood, [...]" 
For similar vitals (heart rate and breathing) we just said, e.g., "Descriptive breathing level".  But I think "Descriptive blood oxygen level" is a bit austere...but I'm fine with it if you prefer the consistency with other vitals.

File: swagger_server/itm/data/metrics/scenarios/metrics-adept-eval-desert.yaml [hunk 2]
```diff
@@ -92,7 +92,7 @@ state:
         mental_status: AGONY
         breathing: NORMAL
         heart_rate: NORMAL
-        spo2: 90
+        spo2: "LOW"
```
Comment (nextcen-dgemoets) [inline swagger_server/itm/data/metrics/scenarios/metrics-adept-eval-desert.yaml:95]: For String controlled vocabulary, we've (almost?) never used double-quotes (although it works just the same).  But I suggest removing them throughout for consistency.

File: swagger_server/itm/data/metrics/test/removed_characters.yaml [hunk 3]
```diff
@@ -56,6 +56,10 @@ state:
         ambulatory: false
         mental_status: UNRESPONSIVE
         breathing: FAST
+        heart_rate: FAST 
+        spo2: "NORMAL"
+        mental_status: UNRESPONSIVE
+        breathing: FAST
         heart_rate: FAST
         spo2: 92
```
Comment (nextcen-dgemoets) [inline swagger_server/itm/data/metrics/test/removed_characters.yaml:64]: Looks like some kind of editing error here.

File: swagger_server/itm/data/metrics/test/removed_characters.yaml [hunk 4]
```diff
@@ -241,10 +244,18 @@ scenes:
           vitals:
             avpu: ALERT
             ambulatory: true
-            mental_status: CALM
-            breathing: NORMAL
+            mental_status: CALM 
+            breathing: NORMAL
```
Comment (nextcen-dgemoets) [inline swagger_server/itm/data/metrics/test/removed_characters.yaml:248]: Extra spaces.

File: swagger_server/itm/data/metrics/test/removed_characters.yaml [hunk 5]
```diff
@@ -241,10 +244,18 @@ scenes:
...
+            spo2: "NORMAL"
+      environment:
+        decision_environment:
+          unstructured: >
+            EVAC has been delayed to 50 minutes out and can only take one patient.
+          aid_delay:
+            - id: small_evac
+              delay: 50
+              max_transport: 1
     action_mapping:
```
Comment (nextcen-dgemoets) [inline swagger_server/itm/data/metrics/test/removed_characters.yaml:259]: Is this a merging issue?  This got added in it seems.

File: swagger_server/itm/data/metrics/test/sample.yaml [hunk 6]
```diff
@@ -108,7 +108,7 @@ state:
         mental_status: UNRESPONSIVE # controlled vocab includes AGONY, CALM, CONFUSED, SHOCK, UPSET, UNRESPONSIVE
         breathing: FAST # controlled vocab includes NORMAL, FAST, RESTRICTED, NONE
         heart_rate: FAST # controlled vocab includes NONE, FAINT, NORMAL, FAST
-        spo2: 92 # a percentage
+        spo2: "NORMAL"
```
Comment (nextcen-dgemoets) [inline swagger_server/itm/data/metrics/test/sample.yaml:111]: The sample YAML also includes the controlled vocab, so please add:
`# controlled vocab includes NONE, LOW, NORMAL`
to all of the `spo2` properties in (only) this file.

File: swagger_server/itm/data/metrics/test/unordered_scenes.yaml [hunk 7]
```diff
@@ -39,6 +39,10 @@ state:
         mental_status: UNRESPONSIVE
         breathing: FAST
         heart_rate: FAST
+        spo2: "NORMAL"
+        mental_status: UNRESPONSIVE
+        breathing: FAST
+        heart_rate: FAST
         spo2: 92 # a percentage
```
Comment (nextcen-dgemoets) [inline swagger_server/itm/data/metrics/test/unordered_scenes.yaml:46]: Similar editing error to above.

**Review by nextcen-dgemoets — CHANGES_REQUESTED** [review]
Looks good.  A couple of editing issues and we're not using double-quotes for simple string vocab in the YAML files, but looks good overall.

File: swagger/swagger.yaml (hunk 1 above)
Comment (Anchita14) [reply swagger/swagger.yaml:1907]: I agree that the description could be made a bit clearer. Other than the few comments made, everything else seems to look fine to me.

File: swagger_server/itm/data/metrics/test/removed_characters.yaml (hunk 5 above)
Comment (Ayal-Ciobotaru) [reply swagger_server/itm/data/metrics/test/removed_characters.yaml:259]: Yes that was a merge issue, environment shouldn't be in there at all, correct?

Comment (nextcen-dgemoets) [reply swagger_server/itm/data/metrics/test/removed_characters.yaml:259]: Correct-- I think it was removed in a recent merge.

**Review by nextcen-dgemoets — APPROVED** [review]
All of the comments have been addressed.  Looks good to me!

File: swagger/swagger.yaml (hunk 1 above)
Comment (nextcen-dgemoets) [reply swagger/swagger.yaml]: @Anchita14   Ayal has updated this PR.  If everything still looks fine to you, can you approve it, and take a look at the [accompanying client PR](https://github.com/NextCenturyCorporation/itm-evaluation-client/pull/44) ?  Thanks!

**Review by Anchita14 — APPROVED** [review]
With the changes, everything seems good now.

**Pull request merged**